### PR TITLE
CORCI-1113 build: Add retry logic to dnf commands

### PR DESCRIPTION
**Pull request by phender**
Adding a common retry/timeout method for any command and using it to
retry dnf install/upgrade commands.

Quick-build: true

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: ci/provisioning/post_provision_config_nodes.sh [hunk 1]
```diff
@@ -73,6 +73,34 @@ dump_repos() {
         done
 }
 
+retry_cmd() {
```
Comment (phender) [inline ci/provisioning/post_provision_config_nodes.sh:76]: Based upon https://github.com/daos-stack/daos/pull/6530/files#diff-c7345873ebf345a4a974743526bb970cc1bd63a8997a452def65e942de7d77e7, I'm wondering if this should just be `timeout_cmd()` without the need for the `if [ "$tries" -gt 0 ]; then; sleep "${DAOS_STACK_RETRY_DELAY_SECONDS:-60}"; fi` code.  @JohnMalmberg, thoughts?

Comment (phender) [reply ci/provisioning/post_provision_config_nodes.sh:76]: Or just simply this:
```
timeout_cmd() {
    local timeout="$1"
    shift
    retry_cmd time timeout "$timeout" "$@"
```

**Comment by phender** [comment]
Failure in CentOS 8 VM stage:
```
[2021-08-31T17:18:55.265Z] + timeout_cmd 5m dnf -y upgrade --exclude 'dpdk,fuse,mercury,daos,daos-*'
[2021-08-31T17:18:55.265Z] + local timeout=5m
[2021-08-31T17:18:55.265Z] + shift
[2021-08-31T17:18:55.265Z] + retry_cmd timeout 5m dnf -y upgrade --exclude 'dpdk,fuse,mercury,daos,daos-*'
[2021-08-31T17:18:55.265Z] + local tries=3
[2021-08-31T17:18:55.265Z] + '[' 3 -gt 0 ']'
[2021-08-31T17:18:55.265Z] + timeout 5m dnf -y upgrade --exclude 'dpdk,fuse,mercury,daos,daos-*'
[2021-08-31T17:18:55.265Z] created by dnf config-manager from https://buil  85 kB/s | 3.0 kB     00:00    
[2021-08-31T17:18:55.265Z] created by dnf config-manager from https://repo  28 kB/s | 2.9 kB     00:00    
[2021-08-31T17:18:55.265Z] created by dnf config-manager from https://repo  20 kB/s | 1.8 kB     00:00    
[2021-08-31T17:18:55.265Z] created by dnf config-manager from https://repo  17 kB/s | 1.5 kB     00:00    
[2021-08-31T17:18:55.265Z] Error: 
[2021-08-31T17:18:55.265Z]  Problem: package hdf5-mpich-tests-1.13.0~rc5-3.el8.x86_64 requires hdf5-mpich2 = 1.13.0~rc5-3.el8, but none of the providers can be installed
[2021-08-31T17:18:55.265Z]   - cannot install both hdf5-mpich-1.13.0~rc5-4.el8_3.x86_64 and hdf5-mpich-1.13.0~rc5-3.el8.x86_64
[2021-08-31T17:18:55.265Z]   - cannot install both hdf5-mpich-1.13.0~rc5-3.el8.x86_64 and hdf5-mpich-1.13.0~rc5-4.el8_3.x86_64
[2021-08-31T17:18:55.265Z]   - cannot install the best update candidate for package hdf5-mpich-tests-1.13.0~rc5-3.el8.x86_64
[2021-08-31T17:18:55.265Z]   - cannot install the best update candidate for package hdf5-mpich-1.13.0~rc5-3.el8.x86_64
[2021-08-31T17:18:55.265Z] (try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
[2021-08-31T17:18:55.265Z] 
[2021-08-31T17:18:55.265Z] real	0m1.960s
[2021-08-31T17:18:55.265Z] user	0m1.542s
[2021-08-31T17:18:55.265Z] sys	0m0.111s
```

**Comment by phender** [comment]
I'm also seeing that the `OPERATIONS_EMAIL` is not set when running `ci/provisioning/post_provision_config_nodes.sh`:
```
[2021-08-31T17:18:55.265Z] + ((  tries--  ))
[2021-08-31T17:18:55.265Z] + mail -s 'Command failed in Not_in_jenkins' -r wolf-105vm1.wolf.hpdd.intel.com@intel.com root@localhost
[2021-08-31T17:18:55.265Z] + '[' 2 -gt 0 ']'
[2021-08-31T17:18:55.265Z] + sleep 60
[2021-08-31T17:18:55.265Z] sh: /usr/bin/procmail: No such file or directory
[2021-08-31T17:18:55.265Z] MDA returned nonzero status 127
[2021-08-31T17:18:55.265Z] sh: /usr/bin/procmail: No such file or directory
[2021-08-31T17:18:55.265Z] MDA returned nonzero status 127
[2021-08-31T17:18:55.265Z] sh: /usr/bin/procmail: No such file or directory
[2021-08-31T17:18:55.265Z] MDA returned nonzero status 127
```

**Comment by phender** [comment]
Example of retry code in operation on Leap15 Functional VM stage:
```
[2021-08-31T21:25:35.163Z] + timeout_cmd 5m dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'
[2021-08-31T21:25:35.163Z] + local timeout=5m
[2021-08-31T21:25:35.163Z] + shift
[2021-08-31T21:25:35.163Z] + retry_cmd timeout 5m dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'
[2021-08-31T21:25:35.163Z] + local tries=3
[2021-08-31T21:25:35.163Z] + '[' 3 -gt 0 ']'
[2021-08-31T21:25:35.163Z] + timeout 5m dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'
[2021-08-31T21:25:35.163Z] Repository 'daos-stack-opensuse-15-x86_64-stable-group' is missing name in configuration, using id.
[2021-08-31T21:25:35.163Z] Repository 'opensuse-15.2-x86_64-group' is missing name in configuration, using id.
[2021-08-31T21:25:35.163Z] Last metadata expiration check: 0:03:38 ago on Tue 31 Aug 2021 08:58:36 PM UTC.
[2021-08-31T21:25:35.163Z] Dependencies resolved.
[2021-08-31T21:25:35.163Z] 
[2021-08-31T21:25:35.163Z]  Problem: package libprotobuf-c-devel-1.3.2-lp152.1.4.x86_64 requires libprotobuf-c1 = 1.3.2, but none of the providers can be installed
[2021-08-31T21:25:35.163Z]   - cannot install both libprotobuf-c1-1.3.3-1.suse.lp152.x86_64 and libprotobuf-c1-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.163Z]   - cannot install both libprotobuf-c1-1.3.2-lp152.1.4.x86_64 and libprotobuf-c1-1.3.3-1.suse.lp152.x86_64
[2021-08-31T21:25:35.163Z]   - cannot install the best update candidate for package libprotobuf-c1-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.163Z]   - cannot install the best update candidate for package libprotobuf-c-devel-1.3.2-lp152.1.4.x86_64
...
[2021-08-31T21:25:35.171Z] --------------------------------------------------------------------------------
[2021-08-31T21:25:35.171Z] Total                                           3.1 MB/s | 361 MB     01:54     
[2021-08-31T21:25:35.171Z] Running transaction check
[2021-08-31T21:25:35.171Z] Transaction check succeeded.
[2021-08-31T21:25:35.171Z] Running transaction test
[2021-08-31T21:25:35.171Z] Transaction test succeeded.
[2021-08-31T21:25:35.171Z] Running transaction
[2021-08-31T21:25:35.171Z] 
[2021-08-31T21:25:35.171Z] real	5m0.075s
[2021-08-31T21:25:35.171Z] user	0m26.193s
[2021-08-31T21:25:35.171Z] sys	0m8.501s
[2021-08-31T21:25:35.171Z] + ((  tries--  ))
[2021-08-31T21:25:35.172Z] + mail -s 'Command failed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/7/' -r wolf-106vm4@intel.com scott.kirvan@intel.com,john.e.malmberg@intel.com,brian.murrell@intel.com,mark.a.olson@intel.com,phillip.henderson@intel.com
[2021-08-31T21:25:35.172Z] + '[' 2 -gt 0 ']'
[2021-08-31T21:25:35.172Z] + sleep 60
[2021-08-31T21:25:35.172Z] + '[' 2 -gt 0 ']'
[2021-08-31T21:25:35.172Z] + timeout 5m dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'
[2021-08-31T21:25:35.172Z] Repository 'daos-stack-opensuse-15-x86_64-stable-group' is missing name in configuration, using id.
[2021-08-31T21:25:35.172Z] Repository 'opensuse-15.2-x86_64-group' is missing name in configuration, using id.
[2021-08-31T21:25:35.172Z] Last metadata expiration check: 0:05:11 ago on Tue 31 Aug 2021 09:03:03 PM UTC.
[2021-08-31T21:25:35.172Z] Dependencies resolved.
[2021-08-31T21:25:35.172Z] 
[2021-08-31T21:25:35.172Z]  Problem: package libprotobuf-c-devel-1.3.2-lp152.1.4.x86_64 requires libprotobuf-c1 = 1.3.2, but none of the providers can be installed
[2021-08-31T21:25:35.172Z]   - cannot install both libprotobuf-c1-1.3.3-1.suse.lp152.x86_64 and libprotobuf-c1-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.172Z]   - cannot install both libprotobuf-c1-1.3.2-lp152.1.4.x86_64 and libprotobuf-c1-1.3.3-1.suse.lp152.x86_64
[2021-08-31T21:25:35.172Z]   - cannot install the best update candidate for package libprotobuf-c1-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.172Z]   - cannot install the best update candidate for package libprotobuf-c-devel-1.3.2-lp152.1.4.x86_64
...
[2021-08-31T21:25:35.197Z] dracut: *** Creating initramfs image file '/boot/initrd-5.3.18-lp152.87-default' done ***
[2021-08-31T21:25:35.197Z] 
[2021-08-31T21:25:35.197Z]   Running scriptlet: cups-2.2.7-lp152.9.9.1.x86_64                      472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: dracut-049.1+suse.203.g8ee14a90-lp152.2.30.1.x86   472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: thin-provisioning-tools-0.7.5-lp152.4.3.1.x86_64   472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: device-mapper-1.02.163-lp152.7.27.1.x86_64         472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: kpartx-0.8.2+166.7501a27-lp152.3.9.1.x86_64        472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: kexec-tools-2.0.20-lp152.4.6.1.x86_64              472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: nvme-cli-1.10-lp152.2.15.1.x86_64                  472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: systemd-presets-common-SUSE-15-lp152.9.9.1.noarc   472/472 
[2021-08-31T21:25:35.198Z]   Running scriptlet: ucode-intel-20210525-lp152.2.17.1.x86_64           472/472
[2021-08-31T21:25:35.198Z] real	12m7.583s
[2021-08-31T21:25:35.198Z] user	1m15.350s
[2021-08-31T21:25:35.198Z] sys	0m18.713s
[2021-08-31T21:25:35.198Z] + ((  tries--  ))
[2021-08-31T21:25:35.198Z] + mail -s 'Command failed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/7/' -r wolf-106vm4@intel.com scott.kirvan@intel.com,john.e.malmberg@intel.com,brian.murrell@intel.com,mark.a.olson@intel.com,phillip.henderson@intel.com
[2021-08-31T21:25:35.198Z] + '[' 1 -gt 0 ']'
[2021-08-31T21:25:35.198Z] + sleep 60
[2021-08-31T21:25:35.198Z] + '[' 1 -gt 0 ']'
[2021-08-31T21:25:35.198Z] + timeout 5m dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'
[2021-08-31T21:25:35.198Z] Repository 'daos-stack-opensuse-15-x86_64-stable-group' is missing name in configuration, using id.
[2021-08-31T21:25:35.198Z] Repository 'opensuse-15.2-x86_64-group' is missing name in configuration, using id.
[2021-08-31T21:25:35.198Z] Last metadata expiration check: 0:18:19 ago on Tue 31 Aug 2021 09:03:03 PM UTC.
[2021-08-31T21:25:35.198Z] Dependencies resolved.
[2021-08-31T21:25:35.198Z] 
[2021-08-31T21:25:35.198Z]  Problem: package libprotobuf-c-devel-1.3.2-lp152.1.4.x86_64 requires libprotobuf-c1 = 1.3.2, but none of the providers can be installed
[2021-08-31T21:25:35.198Z]   - cannot install both libprotobuf-c1-1.3.3-1.suse.lp152.x86_64 and libprotobuf-c1-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.198Z]   - cannot install both libprotobuf-c1-1.3.2-lp152.1.4.x86_64 and libprotobuf-c1-1.3.3-1.suse.lp152.x86_64
[2021-08-31T21:25:35.198Z]   - cannot install the best update candidate for package libprotobuf-c1-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.198Z]   - cannot install the best update candidate for package libprotobuf-c-devel-1.3.2-lp152.1.4.x86_64
[2021-08-31T21:25:35.198Z] ==============================================================================================
[2021-08-31T21:25:35.198Z]  Package          Arch    Version            Repository                                   Size
[2021-08-31T21:25:35.198Z] ==============================================================================================
[2021-08-31T21:25:35.198Z] Skipping packages with conflicts:
[2021-08-31T21:25:35.198Z] (add '--best --allowerasing' to command line to force their upgrade):
[2021-08-31T21:25:35.198Z]  libprotobuf-c1   x86_64  1.3.3-1.suse.lp152 daos-stack-opensuse-15-x86_64-stable-group   68 k
[2021-08-31T21:25:35.198Z] 
[2021-08-31T21:25:35.198Z] Transaction Summary
[2021-08-31T21:25:35.198Z] ==============================================================================================
[2021-08-31T21:25:35.198Z] Skip  1 Package
[2021-08-31T21:25:35.198Z] 
[2021-08-31T21:25:35.198Z] Nothing to do.
[2021-08-31T21:25:35.198Z] Complete!
[2021-08-31T21:25:35.198Z] 
[2021-08-31T21:25:35.198Z] real	0m8.542s
[2021-08-31T21:25:35.198Z] user	0m2.965s
[2021-08-31T21:25:35.198Z] sys	0m0.673s
[2021-08-31T21:25:35.198Z] + return 0
```

A few observations:

- The `sleep 60` does not appear to be working based upon the log timestamps
- The `dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'` command is retried three times as intended, but it doesn't seem like it should have been successful on the third try.
- Would it make sense for us to either log the `${PIPESTATUS[0]}` and/or send an email if the retry logic was able to successfully run the command (or send a more severe titled email if we run out of retries)?

@JohnMalmberg, @brianjmurrell, thoughts?

**Comment by JohnMalmberg** [comment]
The libprotobuf-c1 install was caused by the recent rebuild of the protobuf-c package.  The package name should be libprotobuf-c-devel on SUSE systems, not protobuf-c.

I have removed the offending libprotobuf-c1 RPM from nexus.

**Comment by JohnMalmberg** [comment]
If we get less than 3 e-mails, we can assume that the retry succeeded.
I am wondering if we want only send one e-mail if a retry was needed and if it worked  or not.   Getting additional e-mails for each retry on an install, which is our most common failure just seems to add more noise to our inboxes.

**Comment by brianjmurrell** [comment]
I've asked in the ticket and maybe that is a better place for the discussion, but just a heads up here about the question in the ticket:  Why?  What problem is this trying to solve?  Is it a bug that needs filing upstream?

**Comment by JohnMalmberg** [comment]
We are having intermittent failures in package installs in post provisioning that would be fixed by waiting a minute or so and retrying the operation.

The exact cause of the failures is still being determined, but some of it appear from the local artifact repository responding too slow for a request for a minute, and others seem to be random issues with the networking.

So the action is to try a recovery and also give us a notification that a recovery was needed so that we can monitor how often this is showing up.

**Comment by phender** [comment]
In https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-6545/runs/12/nodes/835/steps/1233/log/?start=0 we hit a case of the retry method in action, but I did not see an email.  Here's the snippet from the log:
```
[2021-09-17T01:09:22.982Z] + rc=124
[2021-09-17T01:09:22.982Z] + ((  attempt++  ))
[2021-09-17T01:09:22.982Z] + '[' 1 -gt 0 ']'
[2021-09-17T01:09:22.982Z] + sleep 60
[2021-09-17T01:09:22.982Z] + '[' 1 -lt 3 ']'
[2021-09-17T01:09:22.982Z] + timeout 5m dnf -y upgrade --exclude 'fuse,fuse-libs,fuse-devel,mercury,daos,daos-*'
[2021-09-17T01:09:22.982Z] Repository 'daos-stack-opensuse-15-x86_64-stable-group' is missing name in configuration, using id.
[2021-09-17T01:09:22.982Z] Repository 'opensuse-15.2-x86_64-group' is missing name in configuration, using id.
[2021-09-17T01:09:22.982Z] Last metadata expiration check: 0:05:41 ago on Fri 17 Sep 2021 12:47:33 AM UTC.
[2021-09-17T01:09:22.982Z] Dependencies resolved.
[2021-09-17T01:09:22.982Z] Nothing to do.
[2021-09-17T01:09:22.982Z] Complete!
[2021-09-17T01:09:22.982Z] 
[2021-09-17T01:09:22.982Z] real	0m3.985s
[2021-09-17T01:09:22.982Z] user	0m3.489s
[2021-09-17T01:09:22.982Z] sys	0m0.401s
[2021-09-17T01:09:22.982Z] + '[' 1 -gt 0 ']'
[2021-09-17T01:09:22.983Z] + send_mail 'Command retry successful in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/12/ after 1 attempts' 'Host:    wolf-106vm3\nCommand: time timeout 5m dnf -y upgrade --exclude fuse,fuse-libs,fuse-devel,mercury,daos,daos-*\nStatus:  0'
[2021-09-17T01:09:22.983Z] + local 'subject=Command retry successful in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/12/ after 1 attempts'
[2021-09-17T01:09:22.983Z] + local 'message=Host:    wolf-106vm3\nCommand: time timeout 5m dnf -y upgrade --exclude fuse,fuse-libs,fuse-devel,mercury,daos,daos-*\nStatus:  0'
[2021-09-17T01:09:22.983Z] + set +x
[2021-09-17T01:09:22.983Z] + return 0
```

**Comment by phender** [comment]
In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/18/ all the tests passed and we received N "Command exceeded <time>s in <stage>" emails:

example:
```
Command exceeded 600s in Functional on Leap 15
from: wolf-100vm3@intel.com

Build: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/18/
Stage: Functional on Leap 15
Host:  wolf-100vm3

Command:  dnf -y upgrade --exclude fuse,fuse-libs,fuse-devel,mercury,daos,daos-*
Real time: 604
```

- wolf-100vm3 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 604
- wolf-100vm2 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 618
- wolf-104vm12 | Command exceeded 600s in Test Leap 15 RPMs | dnf -y upgrade ... | 714
- wolf-100vm5 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 1426
- wolf-100vm7 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 1390
- wolf-100vm8 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 1603
- wolf-100vm6 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 1394
- wolf-100vm9 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 1833
- wolf-100vm4 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade ... | 1699

**Review by brianjmurrell — COMMENTED** [review]
Nothing that should really prevent landing but some suggestions on DRYing out some code and adding some more flexibility.

File: ci/provisioning/post_provision_config_nodes_EL_7.sh [hunk 2]
```diff
@@ -5,31 +5,8 @@ DISTRO_NAME=centos7
...
-        # yum failed for something other than timeout
-        return 1
-    done
-
-    return 1
-}
-
 bootstrap_dnf() {
-    timeout_yum 5m install dnf 'dnf-command(config-manager)'
+    timeout_cmd 5m yum -y install dnf 'dnf-command(config-manager)'
```
[the remark this answers is not in the document]
Comment (phender) [reply ci/provisioning/post_provision_config_nodes_EL_7.sh:9]: Reinstated the original `timeout_yum` logic in `timeout_cmd`.

File: ci/provisioning/post_provision_config_nodes.sh [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 set -eux
 
+: "${DAOS_STACK_RETRY_DELAY_SECONDS:=60}"
+: "${DAOS_STACK_RETRY_COUNT:=3}"
+: "${DAOS_STACK_MONITOR_SECONDS:=600}"
+: "${BUILD_URL:=Not_in_jenkins}"
+: "${STAGE_NAME:=Unknown_Stage}"
+: "${OPERATIONS_EMAIL:=$USER@localhost}"
```
[the remark this answers is not in the document]
Comment (phender) [reply ci/provisioning/post_provision_config_nodes.sh]: Move to new common file.

File: ci/provisioning/post_provision_config_nodes.sh [hunk 4]
```diff
@@ -73,6 +80,70 @@ dump_repos() {
...
+    set +x
+    {
+        echo "Build: $BUILD_URL"
+        echo "Stage: $STAGE_NAME"
+        echo "Host:  $HOSTNAME"
+        echo ""
+        echo -e "$message"
+    } 2>&1 | mail -s "$subject" -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+    set -x
+}
```
[the remark this answers is not in the document]
Comment (phender) [reply ci/provisioning/post_provision_config_nodes.sh]: Suggestion applied.

File: ci/provisioning/post_provision_config_nodes.sh [hunk 5]
```diff
@@ -73,6 +80,70 @@ dump_repos() {
...
+        if [ "$attempt" -gt 0 ]; then
+            sleep "$DAOS_STACK_RETRY_DELAY_SECONDS"
+        fi
+    done
+    if [ "$rc" -ne 0 ]; then
+        send_mail "Command retry failed in $STAGE_NAME after $attempt attempts" \
+                  "Command:  $*\nAttempts: $attempt\nStatus:   $rc"
+    fi
+    return 1
+}
```
[the remark this answers is not in the document]
Comment (phender) [reply ci/provisioning/post_provision_config_nodes.sh]: Common definition of `retry_cmd` is now used.  I did not make the monitor time optional, but instead I added values to the `ci/provisioning/post_provision_config.sh` `retry_cmd` calls.

File: ci/provisioning/post_provision_config.sh [hunk 6]
```diff
@@ -15,34 +15,48 @@ EOF
...
+}
+
 retry_cmd() {
     local command="$1"
     shift
 
-    local tries=$DAOS_STACK_RETRY_COUNT
-    while [ $tries -gt 0 ]; do
+    local attempt=0
+    while [ $attempt -lt $DAOS_STACK_RETRY_COUNT ]; do
```
[the remark this answers is not in the document]
Comment (phender) [reply ci/provisioning/post_provision_config.sh]: Suggestion added.

**Comment by phender** [comment]
In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6545/23/ emails were sent for the following commands:

- wolf-100vm10 | Command exceeded 600s in Test Leap 15 RPMs | dnf -y upgrade | 1201
- wolf-100vm4 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade | 1606
- wolf-100vm5 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade | 1618
- wolf-100vm6 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade | 1498
- wolf-100vm7 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade | 1505
- wolf-100vm8 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade | 1523
- wolf-100vm9 | Command exceeded 600s in Functional on Leap 15 | dnf -y upgrade | 1424

**Review by brianjmurrell — APPROVED** [review]
Great stuff.  Really good implementation of suggestions.

**Review by JohnMalmberg — APPROVED** [review]

**Pull request merged**